solver.rb

```diff
@@ -0,0 +1,23 @@
+class Solver
+  def self.factorial(number)
+    raise ArgumentError, 'n must be non-negative' if number.negative?
+    return 1 if number.zero?
+
+    (1..number).inject(:*)
+  end
+
+  def self.reverse(word)
+    raise ArgumentError, 'Invalid input' if word.class != String
+
+    word.reverse
+  end
+
+  def self.fizzbuzz(number)
+    case number
+    when number % 15 then 'fizzbuzz'
+    when number % 3 then 'fizz'
+    when number % 5 then 'buzz'
+    else number.to_s
+    end
+  end
+end
```

Comment:
- [ ] It's great that you have made all the methods and their tests :+1: but you know that this project is about TDD means you would write your tests before writing any method in `solver.rb` file :thinking: and your commit history will prove if you have followed TDD :+1: how? You would write your tests and make a commit then you would write your methods and make a commit (In this second method kindly don't touch your tests files so we will be sure that you have to write your methods and didn't change your tests) :+1: